src/test/java/io/rcktapp/api/handler/sql/SqlGetHandlerTest.java

```diff
@@ -0,0 +1,120 @@
+package io.rcktapp.api.handler.sql;
+
+import io.forty11.j.J;
+import io.forty11.sql.Sql;
+import io.forty11.web.Url;
+import io.rcktapp.api.*;
+
+import io.rcktapp.api.Collection;
+import io.rcktapp.api.handler.sql.SqlDb;
+import io.rcktapp.api.handler.sql.SqlGetHandler;
+import io.rcktapp.api.service.Service;
+
+import io.rcktapp.rql.Replacer;
+import io.rcktapp.rql.Rql;
+import io.rcktapp.rql.Stmt;
+import io.rcktapp.rql.sql.SqlRql;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.*;
+
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.sql.Array;
+import java.sql.Connection;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class SqlGetHandlerTest {
+
+    @Mock
+    Service service;
+
+    @Mock
+    Api api;
+
+    @Mock
+    Endpoint endpoint;
+
+    @Mock
+    Action action;
+
+    @Mock
+    Chain chain;
+
+    @Mock
+    Response response;
+
+    @Mock
+    Collection collection;
+
+    @Mock
+    Entity entity;
+
+    @Mock
+    Table table;
+
+    @Mock
+    Attribute attribute;
+
+    @Mock
+    Stmt statement;
+
+    @Mock
+    SqlDb sqlDb;
+
+    @Mock
+    SqlRql sqlRql;
+
+    @Mock
+    Connection connection;
+
+    @Mock
+    SqlGetHandler handler;
+
+    @Captor
+    ArgumentCaptor<List> captor;
+
+    @Test
+    public void testSqlGetHandler_properParmasPassedToQueryMethod() throws Exception {
+        Url url = new Url("http://localhost:8080/api/lift/us/plugins/updatexml(rand(),concat(CHAR(126),user(),CHAR(126)),null)");
+
+        Request request = new Request(url, "GET", new HashMap<>(), new HashMap<>(), "");
+        Service.ApiMatch apiMatch = new Service.ApiMatch(api, endpoint, "GET", request.getUrl(), "http://localhost:8080/api/lift/us/", "plugins/updatexml(rand(),concat(CHAR(126),user(),CHAR(126)),null)/");
+
+        request.setApiMatch(apiMatch);
+        collection.setName("plugins");
+
+        mockStatic(Rql.class);
+        mockStatic(Sql.class);
+
+        when(chain.getService()).thenReturn(service);
```

Comment:
Rather than mocking the world, is it possible to just .... start an inversion instance?

Reply:
Wouldn't that not be a unit test at that point?

Reply:
strictly it might be more of an integration test at that point but when testing things like DB you're kinda doing an integration test regardless...

we might, however, be able to get away w/ fewer mocks where some of these things are genuinely Beans or Collections that we could just use setters to populate rather than mocking get calls